crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1739,7 +1740,7 @@ impl From<ExprStarred> for Expr {
 #[derive(Clone, Debug, PartialEq)]
 pub struct ExprName {
     pub range: TextRange,
-    pub id: String,
+    pub id: SmolStr,
```

Comment:
How did you decide where to use `SmolStr`? Should `ExprIpyEscapeCommand`, `StringLiteral`, and `FStringElement` use `SmolStr` too or is it intentional that they do not because they're less likely to be short?

Reply:
I already answered most of this question in another comment, but I didn't mention `ExprIpyEscapeCommand`. I haven't seen many examples of real world use of `ExprIpyEscapeCommand` so I don't know if they are usually short, like `Names` and `Identifiers`. Can't really tell if they are a suitable candidate to use `SmolStr`.